dm/syncer/syncer.go

```diff
@@ -2140,6 +2187,42 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
 	}
 }
 
+func (s *Syncer) checkAndSetSafeModeByBinlogTS(firstBinlogTS int64) error {
+	// see more in https://github.com/pingcap/tiflow/pull/4601#discussion_r814446628
+	duration, err := time.ParseDuration(s.cliArgs.SafeModeDuration)
+	if err != nil {
+		return err
+	}
+	s.firstMeetBinlogTS = &firstBinlogTS
+	exitTS := firstBinlogTS + int64(duration.Seconds())
+	s.exitSafeModeTS = atomic.NewInt64(exitTS)
+	s.tctx.L().Info("safe-mode will disable by task cli args", zap.Int64("ts", exitTS))
+	return nil
+}
+
+func (s *Syncer) checkAndExitSafeModeByBinlogTS(ctx *tcontext.Context, ts int64) error {
+	if s.exitSafeModeTS != nil && ts > s.exitSafeModeTS.Load() {
```

Comment:
no need to use atomic int for exitSafeModeTS? because it's only accessed in main goroutine of syncer